x/btcstaking/keeper/btc_reorg.go

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+)
+
+// HaltIfBtcReorgLargerThanConfirmationDepth safety mechanism to stop the chain in case there is an BTC reorg
+// higher than the BtcConfirmationDepth. In teory this should only happen if the babylon chain goes down for
+// a period longer than (2 * BtcConfirmationDepth * 10min).
+func (k *Keeper) HaltIfBtcReorgLargerThanConfirmationDepth(ctx context.Context) {
+	p := k.btccKeeper.GetParams(ctx)
+
+	largestReorg := k.MustGetLargestBtcReorg(ctx)
+	if largestReorg > p.BtcConfirmationDepth {
+		panic(fmt.Sprintf("Reorg %d is larger than BTC confirmation Depth %d", largestReorg, p.BtcConfirmationDepth))
+	}
+}
+
+// SetLargestBtcReorg sets the new largest BTC block reorg if it is higher than the current
+// value in the store.
+func (k *Keeper) SetLargestBtcReorg(ctx context.Context, newLargestBlockReorg uint32) error {
+	exists, err := k.LargestBtcReorgInBlocks.Has(ctx)
```

Comment:
when does `LargestBtcReorgInBlocks.Has` can return error ? only if db is corrupted ? (if so maybe panic coud be considered here)

Reply:
yes, if encoding fails 
That makes sense, I will also panic if Get returns an error in the cases where we verify if exists it first